specification.md

```diff
@@ -3555,12 +3555,14 @@ The `scheme` property defines the auth scheme to be used. Can be "bearer", "basi
 The `properties` property defines the auth scheme properties information.
 Can be one of ["Basic properties definition"](#basic-properties-definition), ["Bearer properties definition"](#bearer-properties-definition), or ["OAuth2 properties definition"](#oauth2-properties-definition)
 
+Implementations might impose restrictions on the expressions supported to resolve auth properties, depending on the context used. 
```

Comment:
I think this sentence might be more confusing to reader than help. Think this is best to be added if needed in the runtime impl docs.

Reply:
This is part of the effort do identify those constructs, like name. version or so one, where expressions are not applicable. 
In that case, for auth definition, expression support might be limited, up to the implementor.

Reply:
@fjtirado Now that I think of it, I guess it therefore means you'd need to add that comment for `functionDef` `operation` property, which can as well be a runtime expression.

Reply:
yes, in general, support for expressions in function def should be implementor choice, in my opinion.

Reply:
so, we can move this sentence from hre to function definition and that will be it, wdty?

Reply:
i dont think operation should allow to be expression

Reply:
it introduces all kinds of problems

Reply:
of you want different uris to end points like in dev env vs prod then use different external files in each env

Reply:
> i dont think operation should allow to be expression

@tsurdilo We discussed that a long time ago, and you had the opposite take on it. As you said, it's a required feature to have in order to allow different configuration for different environments, resolved at runtime.

Does not change much for me, Synapse supports it anyways, but I think it's a must have feature for real life use cases

Reply:
as you can guess, I do not think either operation or auth definitions used for downloading the operation definition should be expressions. 
The only reason I open the discussion is to avoid hardcoding password in auth definition (using $SECRET there)

Reply:
I agree with @tsurdilo. Operations should be immutable, IMO. This kind of configuration is a matter of infrastructure configuration, not business domain. If we allow this, it will open a huge can of worms.

Reply:
>  If we allow this, it will open a huge can of worms.

yes, and if possible (let me know) id like that can of works to be on the runtime impl side :)

Reply:
To me, conceptually at least, auth definition is part of the operation definition that Ricardo and Tiho think should be "inmutable". Thats why I was reluctant to accept "indiscrimanted" support for expressions in auth def, but, after listening Charles thought, I now think is a grey area. 
We can say nothing and let implementation decide. So if everybody agree Ill remove this sentence and just keep the other modification.